__tests__/integration/streaming_e2e.test.ts

```diff
@@ -32,13 +31,27 @@ describe('streaming e2e', () => {
       __dirname,
       './fixtures/streaming_e2e_data.ndjson'
     )
-
+    const readStream = fs.createReadStream(filename)
+    const jsonTransform = new Transform({
+      transform(data: Buffer, encoding, callback) {
+        data
+          .toString(encoding)
+          .split('\n')
+          .forEach((line) => {
+            if (!line.length) {
+              return
+            } else {
+              const json = JSON.parse(line)
+              this.push(json)
+            }
+          })
+        callback()
+      },
+      objectMode: true,
+    })
```

Comment:
Not the best implementation for this (haven't tested how it behaves with large files), but at least it works without `split2`.